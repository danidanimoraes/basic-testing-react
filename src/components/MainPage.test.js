import React from 'react';
import { shallow, mount, render } from 'enzyme';
import MainPage from './MainPage';
import "../../setupTests";


let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchfield: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
});

it('should render main page', () => {
    expect(wrapper).toMatchSnapshot()
});

it('should filter robots', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [
            {
                id: 1,
                name: 'aaa',
                email: 'a@gmail.com'
            }
        ],
        searchField: 'x',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />)
    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([]);
});
