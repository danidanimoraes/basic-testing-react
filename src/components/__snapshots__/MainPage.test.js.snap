// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render main page 1`] = `ShallowWrapper {}`;
